### ISSUE TITLE
Have the History widget dislpay a separate history for each console

### ISSUE THREAD

**Issue by bcolsen** [spyder-ide/spyder#7155]
Currently the history is mixed for all open consoles it would make the most sense if the history was separated for each console.

From a use case prospective I think this isn't a good idea to have the history mixed and it would be better and simpler to have IPython take care of the history entirely as per the changes in #7141. We can just take what the console gives for it's history and not worry about it...it works great. This would require us to have one history tab per console, but luckily the history widget already has tabs implemented.

Currently, If you change between tabs and run commands in each the history gives you a mess of code that is most likely unassociated and you can tell what commands were executed in each console. The up arrow history at the ipython prompt keeps the history separate and when both sessions are closed the two histories are appended into the main history one console after the other so the code is still in blocks that are still associated.

The hardest part would be to synchronize the tab switching between the consoles and the history.

@ccordoba12  suggest to use a QStackedWidget like the variable explorer does instead of tabs

**Comment by impact27** [comment]
It would be quite easy to do, If you replace "iPython history" in register_client by the name of the console, you have a new tab with the correct name. 
The tab are not automatically switched but this should not be a big problem.
The only issue I would see is that the log loaded from the file would be the same for all log tabs.

**Comment by ccordoba12** [comment]
> The only issue I would see is that the log loaded from the file would be the same for all log tabs.

Yeah, this is ugly. That's why (perhaps) it's better to show a single history instead of many tabs.

**Comment by bcolsen** [comment]
@impact27 I think the idea is a redesigned history widget without tabs but using QStackedWidgets.  In this way the history widget only shows the history of the currently displayed console. This would be similar to the way the variable explorer works currently.

**Comment by ccordoba12** [comment]
Yes, that's the idea. The fact that after a restart the history of all consoles will be mixed up and exactly the same makes me prefer this alternative over the one of one history tab per console.
